aws/src/main/java/org/apache/iceberg/aws/lakeformation/LakeFormationCredentialsCache.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.lakeformation;
+
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.Policy;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+class LakeFormationCredentialsCache
+    implements LoadingCache<String, LakeFormationTemporaryCredentials> {
+  private final LoadingCache<String, LakeFormationTemporaryCredentials> cache;
+  private final Clock clock;
+  private final Duration expiryLeadTimeForRefresh;
+
+  LakeFormationCredentialsCache(LoadingCache<String, LakeFormationTemporaryCredentials> cache) {
+    this.cache = Preconditions.checkNotNull(cache, "Cache cannot be null");
+    this.clock = Clock.systemUTC();
+    this.expiryLeadTimeForRefresh = Duration.ofMillis(TimeUnit.MINUTES.toMillis(1));
+  }
+
+  boolean needsRefresh(LakeFormationTemporaryCredentials credentials) {
+    if (credentials == null) {
+      return true;
+    }
+
+    if (credentials
+        .sessionCredentialsExpiration()
+        .minus(expiryLeadTimeForRefresh)
+        .isBefore(clock.instant())) {
+      return true;
+    }
+    return false;
```

Comment:
Nit: maybe this is cleaner, I'm not super opinionated considering it's a longer statement. 

`return credentials
        .sessionCredentialsExpiration()
        .minus(expiryLeadTimeForRefresh)
        .isBefore(clock.instant())`

Reply:
done. thanks!